Terminal.Gui/Drawing/LineCanvas.cs

```diff
@@ -128,22 +82,19 @@ public Rect Bounds
 
                 for (var i = 1; i < _lines.Count; i++)
                 {
-                    StraightLine line = _lines [i];
-                    Rect lineBounds = line.Bounds;
-                    bounds = Rect.Union (bounds, lineBounds);
-                }
-
-                if (bounds.Width == 0)
-                {
-                    bounds.Width = 1;
+                    bounds = Rect.Union (bounds, _lines [i].Bounds);
                 }
 
-                if (bounds.Height == 0)
+                if (bounds is {Width: 0} or {Height: 0})
                 {
-                    bounds.Height = 1;
+                    bounds = bounds with
+                    {
+                        Width = Math.Clamp (bounds.Width, 1, short.MaxValue),
+                        Height = Math.Clamp (bounds.Height, 1, short.MaxValue)
+                    };
                 }
 
-                _cachedBounds = new Rect (bounds.X, bounds.Y, bounds.Width, bounds.Height);
```

Comment:
It's already a struct. It's not necessary to explicitly call the constructor to make a new one, because it's implicit.